pkg/metrics/metrics_test.go

```diff
@@ -0,0 +1,64 @@
+package metrics
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	dto "github.com/prometheus/client_model/go"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+func TestMetrics(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Metrics Suite")
+}
+
+func findMetric(src []*dto.MetricFamily, query string) *dto.MetricFamily {
+	for _, s := range src {
+		if s.Name != nil && *s.Name == query {
+			return s
+		}
+	}
+	return nil
+}
+
+var _ = Describe("Metrics", func() {
+	value := 12
+	sr := "simple-kmod"
+	state := "templates/0000-buildconfig.yaml"
+	kind := "BuildConfig"
+	name := "simple-kmod-driver-build"
+	namespace := "openshift-special-resource-operator"
+
+	m := New()
+	m.SetSpecialResourcesCreated(value)
+	m.SetCompletedState(sr, state, value)
+	m.SetCompletedKind(sr, kind, name, namespace, value)
+
+	It("correctly passes calls to the collectors", func() {
+		expected := []struct {
+			query string
+			value int
+		}{
+			{createdSpecialResourcesQuery, value},
+			{completedStatesQuery, value},
+			{completedKindQuery, value},
+		}
+
+		data, err := metrics.Registry.Gather()
+		Expect(err).To(BeNil())
+		Expect(data).To(HaveLen(len(expected)))
+
+		for _, e := range expected {
+			m := findMetric(data, e.query)
+			Expect(m).ToNot(BeNil(), fmt.Sprintf("metric for %s could not be find", e.query))
```

Comment:
Typo + it looks like [Gomega accepts Printf-like strings](https://onsi.github.io/gomega/#annotating-assertions) already.

```suggestion
			Expect(m).ToNot(BeNil(), "metric for %s could not be found", e.query)
```